provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -330,7 +342,10 @@ func generateSchema() schema.PackageSpec {
 					},
 					"creationRoleProvider": {
 						// Note: It'd be nice if we could pass the ClusterCreationRoleProvider component here.
-						TypeSpec: schema.TypeSpec{Ref: "#/types/eks:index:CreationRoleProvider"},
+						TypeSpec: schema.TypeSpec{
+							Ref:   "#/types/eks:index:CreationRoleProvider",
+							Plain: true,
+						},
```

Comment:
This is plain in the implementation:

https://github.com/pulumi/pulumi-eks/blob/c6f1fce98fa2683abe7ffbad7f166da99928fb83/nodejs/eks/cluster.ts#L1016-L1020